### DISCUSSION TITLE
Build and test on Linux/ARM64

### DISCUSSION THREAD

**Discussion by odidev** [fltk/fltk#1091] [Show and tell]
## Description 

For aarch64 platform, I have built "FLTK" from source. 

  

## Steps  

Following [STEPS](https://github.com/fltk/fltk/blob/master/README.CMake.txt) were used for the building and installation: 

  
~~~
$ git clone https://github.com/fltk/fltk.git && cd fltk 

$ git checkout release-1.3.3 

$ mkdir build && cd build 

$ cmake .. 

$ make 
~~~
  

## Outcome 

To validate that the build was successful we ran version check. The version info returned matched expected value(s).  

  

All this information is also added to [Software Ecosystem Dashboard for Arm](https://www.arm.com/developer-hub/ecosystem-dashboard/).

**Comment by Albrecht-S** [comment]
@odidev Thank you very much for this info and for your efforts to document the usage of FLTK on aarch64. I converted the original GitHub **Issue** to a **Discussion** because issues are reserved for bug reports and RFE's.

I'm surprised though that you used CMake for FLTK version 1.3.3 which is really ancient (10 years ago). However, I noticed that you (or someone else) writes:

> Fast Light ToolKit (FLTK) works on Arm Linux servers starting from version 1.3.3 released in November 2014.

which seems to explain why you used such an old version in the first place.

**_Note to readers:_** FLTK's CMake support in version 1.3.3 was certainly much less mature than it is now and we recommend FLTK 1.4 ~(currently Git only but to be released soon)~ for real use with CMake.
**_Edit:_** Meanwhile, as of Aug 2025, FLTK 1.4.4 has been released. We recommend to use the latest 1.4.x release and/or to test the current development version 1.5 (Git branch `master`).
For the latest releases please go to https://github.com/fltk/fltk/releases.

**Comment by MatthiasWM** [comment]
Tested 1.4RC1 with FLUID based app Einstein https://github.com/pguyot/Einstein on macOS 14.6.1 on ARM64/M1 and all is very good!
